### ISSUE TITLE
[BUG] Warnings in fvdb-core pytests

### ISSUE THREAD

**Issue by harrism** [openvdb/fvdb-core#260]
When we run pytests, we get the following warnings. The first one is trivial to fix. I'm not sure about the others so opening this issue for discussion.

```
unit/test_types.py:414
  /home/mharris/github/fvdb-core/tests/unit/test_types.py:414: SyntaxWarning: "is not" with 'int' literal. Did you mean "!="?
    if inp is not 1:

tests/unit/test_dense_interface.py::TestDenseInterfaceBatch::test_write_to_dense_xyzc_czyx_rand_point_grid_0_cpu
tests/unit/test_dense_interface.py::TestDenseInterfaceBatch::test_write_to_dense_xyzc_czyx_rand_point_grid_1_cuda
tests/unit/test_dense_interface.py::TestDenseInterfaceBatch::test_write_to_dense_xyzc_czyx_rand_point_grid_2_cpu
tests/unit/test_dense_interface.py::TestDenseInterfaceBatch::test_write_to_dense_xyzc_czyx_rand_point_grid_3_cuda
tests/unit/test_dense_interface.py::TestDenseInterfaceBatch::test_write_to_dense_xyzc_czyx_rand_point_grid_4_cpu
tests/unit/test_dense_interface.py::TestDenseInterfaceBatch::test_write_to_dense_xyzc_czyx_rand_point_grid_5_cuda
  /home/mharris/github/fvdb/tests/unit/test_dense_interface.py:370: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.detach().clone() or sourceTensor.detach().clone().requires_grad_(True), rather than torch.tensor(sourceTensor).

tests/unit/test_dense_interface.py::TestDenseInterfaceSingle::test_write_to_dense_xyzc_czyx_rand_point_grid_0_cpu
tests/unit/test_dense_interface.py::TestDenseInterfaceSingle::test_write_to_dense_xyzc_czyx_rand_point_grid_1_cuda
tests/unit/test_dense_interface.py::TestDenseInterfaceSingle::test_write_to_dense_xyzc_czyx_rand_point_grid_2_cpu
tests/unit/test_dense_interface.py::TestDenseInterfaceSingle::test_write_to_dense_xyzc_czyx_rand_point_grid_3_cuda
tests/unit/test_dense_interface.py::TestDenseInterfaceSingle::test_write_to_dense_xyzc_czyx_rand_point_grid_4_cpu
tests/unit/test_dense_interface.py::TestDenseInterfaceSingle::test_write_to_dense_xyzc_czyx_rand_point_grid_5_cuda
  /home/mharris/github/fvdb/tests/unit/test_dense_interface.py:809: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.detach().clone() or sourceTensor.detach().clone().requires_grad_(True), rather than torch.tensor(sourceTensor).

tests/unit/test_inject.py: 12 warnings
  /home/mharris/github/fvdb/tests/unit/test_inject.py:664: UserWarning: The .grad attribute of a Tensor that is not a leaf Tensor is being accessed. Its .grad attribute won't be populated during autograd.backward(). If you indeed want the .grad field to be populated for a non-leaf Tensor, use .retain_grad() on the non-leaf Tensor. If you access the non-leaf Tensor by mistake, make sure you access the leaf Tensor instead. See github.com/pytorch/pytorch/pull/30531 for more informations. (Triggered internally at /home/conda/feedstock_root/build_artifacts/libtorch_1760296002163/work/build/aten/src/ATen/core/TensorBody.h:489.)

tests/unit/test_inject.py: 12 warnings
  /home/mharris/github/fvdb/tests/unit/test_inject.py:701: UserWarning: The .grad attribute of a Tensor that is not a leaf Tensor is being accessed. Its .grad attribute won't be populated during autograd.backward(). If you indeed want the .grad field to be populated for a non-leaf Tensor, use .retain_grad() on the non-leaf Tensor. If you access the non-leaf Tensor by mistake, make sure you access the leaf Tensor instead. See github.com/pytorch/pytorch/pull/30531 for more informations. (Triggered internally at /home/conda/feedstock_root/build_artifacts/libtorch_1760296002163/work/build/aten/src/ATen/core/TensorBody.h:489.)

tests/unit/test_inject.py: 12 warnings
  /home/mharris/github/fvdb/tests/unit/test_inject.py:501: UserWarning: The .grad attribute of a Tensor that is not a leaf Tensor is being accessed. Its .grad attribute won't be populated during autograd.backward(). If you indeed want the .grad field to be populated for a non-leaf Tensor, use .retain_grad() on the non-leaf Tensor. If you access the non-leaf Tensor by mistake, make sure you access the leaf Tensor instead. See github.com/pytorch/pytorch/pull/30531 for more informations. (Triggered internally at /home/conda/feedstock_root/build_artifacts/libtorch_1760296002163/work/build/aten/src/ATen/core/TensorBody.h:489.)

tests/unit/test_inject.py: 12 warnings
  /home/mharris/github/fvdb/tests/unit/test_inject.py:530: UserWarning: The .grad attribute of a Tensor that is not a leaf Tensor is being accessed. Its .grad attribute won't be populated during autograd.backward(). If you indeed want the .grad field to be populated for a non-leaf Tensor, use .retain_grad() on the non-leaf Tensor. If you access the non-leaf Tensor by mistake, make sure you access the leaf Tensor instead. See github.com/pytorch/pytorch/pull/30531 for more informations. (Triggered internally at /home/conda/feedstock_root/build_artifacts/libtorch_1760296002163/work/build/aten/src/ATen/core/TensorBody.h:489.)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

**Comment by fwilliams** [comment]
For that one it's probably worth doing what the warning says and using `.detach().clone()`

**Comment by mvanhorn** [comment]
Implemented in https://github.com/openvdb/fvdb-core/pull/654 - switches test regex strings to raw literals and uses `tensor.clone()` for the copy-construct sites.
